addons/udes_stock/models/stock_move.py

```diff
@@ -545,3 +545,29 @@ def _action_assign(self):
         """
         super()._action_assign()
         return self
+
+    def _action_confirm(self, merge=True, merge_into=False):
+        """
+        Assign procurement group for moves where picking type create procurement groups is enabled
+        """
+        Procurement = self.env["procurement.group"]
+
+        for orderpoint_moves in self.filtered(
```

Comment:
We can just loop over the moves that have `mv.picking_type_id.u_create_procurement_group and not mv.group_id` and then have an if to decide the name, then we only need 
```

Reply:
Pushed suggested changes